scripts/travis-deploy.sh

```diff
@@ -9,8 +9,16 @@
 
 tag=$1
 
-python3 codalab_service.py build --version $tag --pull --push
+# Check if ENV variable CODALAB_DOCKER_USERNAME is set. If not, it means this build was triggered
+# by an external user. Then we shouldn't push the docker image to docker hub.
+if [[ -z "${CODALAB_DOCKER_USERNAME}" ]]; then
```

Comment:
It might look better to do something like
`PUSH_FLAG=$([ -z "${CODALAB_DOCKER_USERNAME}" ] || echo "--push")`

to be consistent with the code in `.travis.yml`.